internal/utils/grpc.go

```diff
@@ -16,10 +19,48 @@ func CloseClientConnection(connection *grpc.ClientConn) {
 }
 
 func GrpcDialWithInsecure(serviceName string, serviceURL string) (*grpc.ClientConn, error) {
-	cc, err := grpc.Dial(serviceURL, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	kacp := keepalive.ClientParameters{
+		Time:                60 * time.Second, // ping the server every ~60 seconds.
+		Timeout:             20 * time.Minute, // wait up to 20 minutes for a packet to be acknowledged.
+		PermitWithoutStream: true,             // send pings if there are no active streams (RPCs).
+	}
+
+	cc, err := grpc.Dial(serviceURL, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithKeepaliveParams(kacp))
 	if err != nil {
 		return nil, fmt.Errorf("could not connect to %s: %w", serviceName, err)
 	}
 
 	return cc, nil
 }
+
+// GrpcDialWithRetryAndBackoff creates an insecure gRPC connection to serviceURL
+// After successfully connected, any RPC calls made from this connection also have a retry
+// policy of ~10 minutes after which an error is returned that it couldn't connect to the service.
+func GrpcDialWithRetryAndBackoff(serviceName, serviceURL string) (*grpc.ClientConn, error) {
+	kacp := keepalive.ClientParameters{
+		Time:                60 * time.Second, // ping the server every ~60 seconds.
+		Timeout:             20 * time.Minute, // wait up to 20 minutes for a packet to be acknowledged.
+		PermitWithoutStream: true,             // send pings if there are no active streams (RPCs).
+	}
+
+	// retry policy for RPC calls.
+	opts := []grpc_retry.CallOption{
+		grpc_retry.WithBackoff(grpc_retry.BackoffLinearWithJitter(45*time.Second, 0.1)),
+		grpc_retry.WithMax(15),
```

Comment:
Ditto, arent 15 retries a bit too much?

Reply:
15 every 45 seconds is ~11 mins, which I think is okay-ish, we could half it to 6 minutes I would be okay either way

Reply:
I would probably prefer to halve it if thats okay